scripts/release/test_package.py

```diff
@@ -0,0 +1,53 @@
+from pathlib import Path
+import subprocess
+from tempfile import TemporaryDirectory
+import venv
+
+
+def create_venv(parent_path):
+    venv_path = parent_path / 'package-smoke-test'
+    venv.create(venv_path, with_pip=True)
+    subprocess.run([venv_path / 'bin' / 'pip', 'install', '-U', 'pip', 'setuptools'], check=True)
+    return venv_path
+
+
+def find_wheel(project_path):
+    wheels = list(project_path.glob('dist/*.whl'))
+
+    if len(wheels) != 1:
+        raise Exception(
+            f"Expected one wheel. Instead found: {wheels} in project {project_path.absolute()}"
+        )
+
+    return wheels[0]
+
+
+def install_wheel(venv_path, wheel_path, extras=tuple()):
+    if extras:
+        extra_suffix = f"[{','.join(extras)}]"
+    else:
+        extra_suffix = ""
+
+    subprocess.run(
+        [
+            venv_path / 'bin' / 'pip',
+            'install',
+            f"{wheel_path}{extra_suffix}"
+        ],
+        check=True,
+    )
+
+
+def test_install_local_wheel():
+    temporary_dir = TemporaryDirectory()
```

Comment:
If you use the context manager API here then the tempdir will still get cleaned up even if/when this script errors out.  If you're intentionally keeping it around it might be good to add a comment indicating as much.

Reply:
Ah, thanks. I was originally doing a `yes/no` prompt to keep it around for inspection. When I refactored back to this current approach, I forgot to go back to the context manager.